### ISSUE TITLE
`toKebab` does not work everywhere

### ISSUE THREAD

**Issue by SiddharthShyniben** [zellwk/javascript#14]
The toKebab function does not work in every case. Here are some examples:

```js
function toKebab (string) {
  return string
    .split('')
    .map((letter, index) => {
      if (/[A-Z]/.test(letter)) {
        return ` ${letter.toLowerCase()}`
      }
      return letter
    })
    .join('')
    .trim()
    .replace(/[_\s]+/g, '-')
}

let samples = [
  "ThereIsWay_too  MuchCGIInFilms These-days",
  "UnicodeCanBeCAPITALISEDTooYouKnow",
  "CAPITALLetters at the StartOfAString_work_too",
  "As_they_DoAtTheEND",
  "BitteWerfenSie-dieFußballeInDenMüll",
  "IchHabeUberGesagtNichtÜber",
  "2BeOrNot2Be",
  "ICannotBelieveThe100GotRenewed. It-isSOOOOOOBad"
];

samples.forEach(sample => console.log(toKebab(sample)))
```

1. `ThereIsWay_too  MuchCGIInFilms These-days` becomes `there-is-way-too-much-c-g-i-in-films-these-days`
2. `IchHabeUberGesagtNichtÜber` becomes `ich-habe-uber-gesagt-nichtÜber`
3. `ICannotBelieveThe100GotRenewed. It-isSOOOOOOBad` becomes `i-cannot-believe-the100-got-renewed.-it-is-s-o-o-o-o-o-o-bad`

--- 

Why don't you try this:
```js
// From https://stackoverflow.com/a/62527131/15511745
function toKebab(string) {
    return string.replace(/([^[\p{L}\d]+|(?<=[\p{Ll}\d])(?=\p{Lu})|(?<=\p{Lu})(?=\p{Lu}[\p{Ll}\d])|(?<=[\p{L}\d])(?=\p{Lu}[\p{Ll}\d]))/gu, '-'
}
```

**Comment by zellwk** [comment]
Thanks for notifying me regarding this and sorry for the lack of response until now. I never thought about continuous capitals until I saw this.

**Comment by jonnybarnes** [comment]
Just FYI, Safari on macOS does not like the regex, it throws a script error

**Comment by zellwk** [comment]
@jonnybarnes Ooh this is interesting. I'l have to try it out before I can confirm. May take a while though!

**Comment by zellwk** [comment]
Tested it, but it seems to work.

**Comment by zellwk** [comment]
Reopening this issue because `toKebab` doesn't work on Safari. Just managed to test this after so long. I'm working on a fix for this.

**Comment by zellwk** [comment]
Closing because this is fixed in v3.0.3
